src/record/api.rs

```diff
@@ -50,19 +77,19 @@ pub enum Row {
   Bytes(ByteArray),
   Timestamp(u64), // Timestamp with milliseconds
   // Complex types
-  Group(Vec<(String, Row)>), // Struct, child elements are tuples of field-value pairs
-  List(Vec<Row>), // List of elements
-  Map(Vec<(Row, Row)>) // List of key-value pairs
+  Group(Row), // Struct, child elements are tuples of field-value pairs
+  List(Vec<RowField>), // List of elements
```

Comment:
I guess when it is list of structs, it will `List(Group(Row), Group(Row), ...)`. Is that right?

Reply:
Yes it will be something like `List(Vec(Group(Row), Group(Row),...))`, and the array interface that I proposed above should have something like `getStruct(rowIndex)` which will return a `Row`.

Reply:
Great, thanks.

Would it be beneficial to also make `List` and `Map` separate, similar to `Row`, so they can define their own set of methods to get length, elements by index, keys and values?

Reply:
I'm thinking about to replace `List<Vec<Row>>` to something like `List<Array>` (we'll need better naming on this) where the `Array` implements the public APIs. Similar for `Map<Vec<Row, Row>>`.  What do you think?

Reply:
I remember you suggested that some time ago. Yes, sounds like a good idea. You could consider calling them `Struct`, `Array`, `Map`, `Field`, and we would return `Struct` for a record, so it aligns with `get_struct`, `get_array`, and `get_map` methods.